tests/unit/schema/test_api.py

```diff
@@ -2602,3 +2604,215 @@ def test_has_iam_mixin():
+            selector="google.example.v1beta1.SomeExample.Example3",
+            auto_populated_fields=[],
+            long_running=client_pb2.MethodSettings.LongRunning(),
+        ),
+        "google.example.v1beta1.SomeExample.Example4": client_pb2.MethodSettings(
+            selector="google.example.v1beta1.SomeExample.Example4",
+            auto_populated_fields=[],
+            long_running=client_pb2.MethodSettings.LongRunning(),
+        ),
+        "google.example.v1beta1.SomeExample.Example5": client_pb2.MethodSettings(
+            selector="google.example.v1beta1.SomeExample.Example5",
+            auto_populated_fields=[],
+            long_running=client_pb2.MethodSettings.LongRunning(),
+        ),
+    }
+
+    # Test that ValueError is raised when the rpc in publishing.method_settings.selector is not a unary one
+    opts = Options(
+        service_yaml_config={
+            "apis": [
+                {"name": "google.example.v1beta1.SomeExample.Example1"},
+            ],
+            "publishing": {
+                "method_settings": [
+                    {
+                        "selector": "google.example.v1beta1.DoesNotExist.Example1",
+                        "auto_populated_fields": [
+                            "squid",
+                            "mollusc",
+                        ],
+                    },
+                    {
+                        "selector": "google.example.v1beta1.SomeExample.Example1",
+                        "auto_populated_fields": [
+                            "doesnotexist",
+                            "mollusc",
+                            "octopus",
+                        ]
+                    },
+                    {
+                        "selector": "google.example.v1beta1.SomeExample.Example2",
+                        "auto_populated_fields": [
+                            "squid",
+                            "mollusc",
+                        ],
+                    },
+                    {
+                        "selector": "google.example.v1beta1.SomeExample.Example3",
+                        "auto_populated_fields": [
+                            "squid",
+                            "mollusc",
+                        ],
+                    }
+                ]
+            },
+        }
+    )
+    api_schema = api.API.build(fd, "google.example.v1beta1", opts=opts)
+
+    exception_message = """Fields cannot be automatically populated in the top level \
+request message for selectors: \['google.example.v1beta1.DoesNotExist.Example1', \
+'google.example.v1beta1.SomeExample.Example1', \
+'google.example.v1beta1.SomeExample.Example2', \
+'google.example.v1beta1.SomeExample.Example3'\]. Selector \
+google.example.v1beta1.DoesNotExist.Example1 is not a valid method in this API. \
+Field `doesnotexist` is not in the top level request message. Field `octopus` is not \
+of type string. Field `octopus` is a required field. Field `octopus` is not a UUID4 \
```

Comment:
Field `octopus` is testing several failing conditions at once. I think we we should test fields with one failing condition at a time. I think a better, accepted practice is to test simple cases and assume if those work then combinations ought to work too, rather than test a combination case and assume that if it works, its constituent parts will then work by themselves.

(The full coverage solution is to test all combinations, but I think that's overkill here.)

Reply:
Fixed in 5a2d33b8b160a246945e3752e145878581a351b1